vault/resource_audit.go

```diff
@@ -0,0 +1,125 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+func auditResource() *schema.Resource {
+	return &schema.Resource{
+		Create: auditWrite,
+		Read:   auditRead,
+		Delete: auditDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Path in which to enable the audit device",
+			},
+
+			"type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Type of the audit device, such as 'file'",
+			},
+
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Required:    false,
+				ForceNew:    true,
+				Description: "Human-friendly description of the audit device",
+			},
+
+			"options": {
+				Type:        schema.TypeMap,
```

Comment:
Can you add the element type to this just to be explicit (I realize its not done in other places in this provider):

```golang
Elem: &schema.Schema{Type: schema.TypeString},
```